deploy/utils/validate_deployments.py

```
@@ -0,0 +1,422 @@
+        dgd_docs = [
+            (i, doc)
+            for i, doc in enumerate(docs)
+            if doc and doc.get("kind") == "DynamoGraphDeployment"
+        ]
+
+        if not dgd_docs:
+            return True, []  # No DGD resources, skip validation
+
+        # Validate each DynamoGraphDeployment
+        errors = []
+        for doc_idx, doc in dgd_docs:
```

Ruff (0.14.4)

Loop control variable `doc_idx` not used within loop body

Rename unused `doc_idx` to `_doc_idx`

(B007)